lib/constructs/identity-provider.ts

```diff
@@ -240,6 +240,14 @@ export class CognitoIdentityProvider extends Construct implements IIdentityProvi
     app.configure(client.clientId, client.secret, this.domain.baseUrl({ fips: true }));
     return client;
   }
+
+  public discoveryUrl(): string {
+    // Breaking up the variables is the most readble to make this work without the string being too long
+    const region = cdk.Aws.REGION;
+    const suffix = cdk.Aws.URL_SUFFIX;
+    const userPoolId = this.userPool.userPoolId;
+    return `https://cognito-idp-fips.${region}.${suffix}/${userPoolId}/.well-known/openid-configuration`;
```

Comment:
Why do you need `${suffix}` isn't it always going to be `amazonaws.com`?

Reply:
Not entirely sure what it _would_ be in other partitions https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html#cfn-pseudo-param-urlsuffix

Reply:
Cognito doesn't currently exist in any other partitions that do/would have a different URL suffix but it may some day and a deployment of our application should work in such a partition.